src/scancode/agent/scancode_wrapper.hpp

```diff
@@ -27,8 +27,7 @@
 
 using namespace std;
 
-string scanFileWithScancode(const State& state, const fo::File& file);
-string scanFileWithScancode(string filename);
+void scanFileWithScancode(const State& state, string file_location, string outputFile);
```

Comment:
What will happen if someone now tries to run scancode agent from CLI like `/usr/local/share/fossology/scancode/agent/scancode file_to_scan.py`? Have you covered it in `main()` somehow?

Reply:
I see now the function `string scanFileWithScancode(string filename);` was never implemented and the CLI scan was not supported. Thus, it is not in the scope of this PR.
If you are interested to work on it, I will request to open an issue and work on a separate PR.